test/distributed/org/apache/cassandra/distributed/test/ReadRepairEmptyRangeTombstonesTest.java

```diff
@@ -0,0 +1,355 @@
+        {
+            for (String query : queries)
+                cluster.schemaChange(withKeyspace(query));
+            return this;
+        }
+
+        Tester createTable(String createTable)
+        {
+            return schemaChange(String.format(createTable, qualifiedTableName, reverse ? "DESC" : "ASC", strategy));
+        }
+
+        Tester executeInternal(int node, String query, Object... boundValues)
+        {
+            return executeInternal(cluster.get(node), query, boundValues);
+        }
+
+        Tester executeInternal(IInvokableInstance node, String query, Object... boundValues)
+        {
+            node.executeInternal(String.format(query, qualifiedTableName), boundValues);
+            return this;
+        }
+
+        private Object[][] queryDistributed(String query, Object... boundValues)
+        {
+            String formattedQuery = String.format(query, qualifiedTableName);
+            ICoordinator coordinator = cluster.coordinator(this.coordinator);
+            return paging
+                   ? Iterators.toArray(coordinator.executeWithPaging(formattedQuery, ALL, 1, boundValues), Object[].class)
+                   : coordinator.execute(formattedQuery, ALL, boundValues);
+        }
+
+        private Tester assertRowsDistributed(String query, long expectedRepaired, Object[]... expectedRows)
+        {
+            // run the query in the coordinator recording the increase in repaired rows metric
+            long actualRepaired = repairedBlockingCount();
+            Object[][] actualRows = queryDistributed(query);
+            actualRepaired = repairedBlockingCount() - actualRepaired;
+
+            // verify the returned rows
+            if (reverse)
+                expectedRows = reverse(expectedRows);
+            AssertUtils.assertRows(actualRows,expectedRows);
+
+            // verify the number of repaired rows
+            if (strategy == ReadRepairStrategy.NONE)
+                expectedRepaired = 0;
+            assertEquals(String.format("Expected %d repaired rows, but found %d", expectedRepaired, actualRepaired),
+                         expectedRepaired, actualRepaired);
+
+            return this;
+        }
+
+        private Tester assertRowsInternal(String query, Object[]... expectedRows)
+        {
+            String formattedQuery = String.format(query, qualifiedTableName);
+
+            if (strategy == ReadRepairStrategy.NONE)
+                expectedRows = EMPTY_ROWS;
+            else if (reverse)
+                expectedRows = reverse(expectedRows);
+
+            AssertUtils.assertRows(cluster.get(2).executeInternal(formattedQuery), expectedRows);
+            return this;
+        }
+
+        private Object[][] reverse(Object[][] rows)
+        {
+            Object[][] reversed = ArrayUtils.clone(rows);
+            ArrayUtils.reverse(reversed);
+            return reversed;
+        }
+
+        private long repairedBlockingCount()
```

Comment:
@adelapena We're starting to have some common methods between the test classes. A base test class built on `TestBaseImpl` or an abstract version of `Tester` *might* make sense, but I'm not advocating either one in particular.

Reply:
Good idea, I have placed the common stuff in a new `ReadRepairTester` class, so the testers in `ReadRepairQueryTypesTest` and `ReadRepairEmptyRangeTombstonesTest` extend it.